jumanji/training/setup_train.py

```diff
@@ -88,7 +89,24 @@ def setup_logger(cfg: DictConfig) -> Logger:
 
 
 def _make_raw_env(cfg: DictConfig) -> Environment:
-    return jumanji.make(cfg.env.registered_version)
+    try:
+        env = jumanji.make(cfg.env.registered_version)
+    except ValueError as error:
+        if (
+            "Unregistered environment" in str(error)
+            and cfg.env.name not in "upgraded_bin_pack"
+        ):
+            raise ValueError(
+                "Unregistered environment setup not possible for any other argument"
+                f"other than bin_pack, env requested is {cfg.env.name}."
+            )
+        env_settings_dict = {}
+        try:
+            env_settings_dict = {**cfg.env.env_settings}
+        except Exception:
```

Comment:
This try except is very general and might not be catching the right thing. It might be better to check the config for whether you want to apply the constraint or not and do an if based on that

Reply:
I fixed this.